src/identical-placeholders.js

```diff
@@ -0,0 +1,184 @@
+const get = require('lodash.get');
+const diff = require('jest-diff');
+const requireNoCache = require('./util/require-no-cache');
+const getTranslationFileSource = require('./util/get-translation-file-source');
+const deepForOwn = require('./util/deep-for-own');
+
+const sortAstNodes = (a, b) => `${a.type}${a.value}`.localeCompare(`${b.type}${b.value}`);
+
+const compareAst = (astA, astB) => {
+  // Skip raw text
+  const astAFiltered = astA.filter(a => a.type !== 0).sort(sortAstNodes);
+  const astBFiltered = astB.filter(a => a.type !== 0).sort(sortAstNodes);
+
+  if (astAFiltered.length !== astBFiltered.length) {
+    return false;
+  }
+
+  if (astAFiltered.length === 0) {
+    return true;
+  }
+
+  return astAFiltered.every((elementA, index) => {
+    const elementB = astBFiltered[index];
+
+    // Type and value should match for each placeholder
+    if (elementA.type !== elementB.type || elementA.value !== elementB.value) {
+      return false;
+    }
+
+    // Compare nested elements for type 5 (select) and 6 (plural)
+    if (elementA.type === 5 || elementA.type === 6) {
+      const elementAOptions = Object.keys(elementA.options).sort();
+      const elementBOptions = Object.keys(elementB.options).sort();
+      if (
+        elementAOptions.length !== elementBOptions.length ||
+        elementAOptions.join('|') !== elementBOptions.join('|')
+      ) {
+        return false;
+      }
+
+      return elementAOptions.every(o =>
+        compareAst(elementA.options[o].value, elementB.options[o].value));
+    }
+
+    // Compare children for type 8 (rich text)
+    if (elementA.type === 8) {
+      return compareAst(elementA.children, elementB.children);
+    }
+
+    return true;
+  });
+};
+
+const identicalPlaceholders = (context, source, sourceFilePath) => {
+  const { options, settings } = context;
+  const { filePath } = options[0] || {};
+
+  if (!filePath) {
+    return [
+      {
+        message: '"filePath" rule option not specified.',
+        loc: {
+          start: {
+            line: 0,
+            col: 0
+          }
+        }
+      }
+    ];
+  }
+
+  // skip comparison with reference file
+  if (filePath === sourceFilePath) {
+    return [];
+  }
+
+  let referenceTranslations = null;
+  let sourceTranslations = null;
+  try {
+    referenceTranslations = requireNoCache(filePath);
+    sourceTranslations = JSON.parse(source);
+  } catch (e) {
+    // don't return any errors
+    // will be caught with the valid-json rule.
+    return [];
+  }
+
+  const ignorePaths = settings['i18n-json/ignore-keys'] || [];
+  const invalidMessages = [];
+
+  deepForOwn(
+    referenceTranslations,
+    (referenceTranslation, _, path) => {
+      if (typeof referenceTranslation === 'string') {
+        const sourceTranslation = get(sourceTranslations, path);
+        if (sourceTranslation) {
+          let referenceAst;
+          let sourceAst;
+          try {
+            referenceAst = parse(referenceTranslation);
```

Comment:
Will this lead to the source reference having to get re-parsed for every translation? Perhaps you could memoize the parsed AST per path.

Reply:
The `sourceTranslation` and `referenceTranslation` refer to a single translation based on the [path](https://github.com/dhaldar-godaddy/eslint-plugin-i18n-json/blob/identical-placeholders/src/identical-placeholders.js#L96). They will be parsed just once.